homeassistant/components/sensoterra/sensor.py

```diff
@@ -0,0 +1,152 @@
+    ),
+    ProbeSensorType.RSSI: SensorEntityDescription(
+        key=ProbeSensorType.RSSI,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SensoterraConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up Sensoterra sensor."""
+
+    coordinator = entry.runtime_data
+
+    @callback
+    def _async_add_devices(probes: list[Probe]) -> None:
+        aha = coordinator.async_contexts()
+        current_sensors = set(aha)
+        async_add_devices(
+            SensoterraEntity(
+                coordinator,
+                probe,
+                sensor,
+                SENSORS[ProbeSensorType[sensor.type]],
+            )
+            for probe in probes
+            for sensor in probe.sensors()
+            if sensor.type is not None
+            and sensor.type.lower() in SENSORS
+            and sensor.id not in current_sensors
+        )
+
+    coordinator.add_devices_callback = _async_add_devices
+
+    _async_add_devices(coordinator.data)
+
+
+class SensoterraEntity(CoordinatorEntity[SensoterraCoordinator], SensorEntity):
+    """Sensoterra sensor like a soil moisture or temperature sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: SensoterraCoordinator,
+        probe: Probe,
+        sensor: Sensor,
+        entity_description: SensorEntityDescription,
+    ) -> None:
+        """Initialize entity."""
+        super().__init__(coordinator, context=sensor.id)
+
+        self._attr_unique_id = sensor.id
+        self._attr_translation_placeholders = {
+            "depth": "?" if sensor.depth is None else str(sensor.depth)
+        }
+
+        self.entity_description = entity_description
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, probe.serial)},
+            name=probe.name,
+            model=probe.sku,
+            manufacturer="Sensoterra",
+            serial_number=probe.serial,
+            suggested_area=probe.location,
+            configuration_url=CONFIGURATION_URL,
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        sensor = self.coordinator.get_sensor(self._attr_unique_id)
+        if sensor is None:
+            self._attr_native_value = None
+            return False
+
+        self._attr_native_value = sensor.value
+
+        if sensor.timestamp is None:
+            return False
+
+        # Expire sensor if no update within the last few days.
+        expiration = datetime.now(UTC) - timedelta(days=SENSOR_EXPIRATION_DAYS)
+        return sensor.timestamp >= expiration
```

Comment:
This is a property, this should not have side effects. You should split this up into 2 properties. One for native value and one for available.

If you feel extra fancy, you can add a property for `sensor` and then you can easily access `self.sensor` to avoid having to write `self.coordinator.get_sensor(...)` in both properties